packages/browser-pool/src/browser-pool.ts

```diff
@@ -327,6 +327,24 @@ export class BrowserPool<
             fingerprintOptions = {},
         } = options;
 
+        const firstPluginConstructor = browserPlugins[0].constructor as typeof BrowserPlugin;
+
+        for (let i = 1; i < browserPlugins.length; i++) {
+            const providedPlugin = browserPlugins[i];
+
+            if (!(providedPlugin instanceof firstPluginConstructor)) {
+                const firstPluginName = firstPluginConstructor.name;
+                const providedPluginName = (providedPlugin as BrowserPlugin).constructor.name;
+
+                throw new Error(
+                    [
+                        `Browser plugin at index ${i} (${providedPluginName}) is not an instance of the same plugin`,
+                        `as the first plugin provided (${firstPluginName}).`,
+                    ].join(' '),
```

Comment:
this looks weird :] i'd just add eslint ignore comment here and keep it all on single line